airflow-core/src/airflow/configuration.py

```diff
@@ -2210,28 +2208,73 @@ def get_custom_secret_backend(worker_mode: bool = False) -> BaseSecretsBackend |
     return secrets_backend_cls(**backend_kwargs)
 
 
+def get_importable_secret_backend(class_name: str | None) -> BaseSecretsBackend | None:
+    """Get secret backend defined in the given class name."""
+    if class_name is not None:
+        secrets_backend_cls = import_string(class_name)
+        return secrets_backend_cls()
+    return None
+
+
 def initialize_secrets_backends(
-    default_backends: list[str] = DEFAULT_SECRETS_SEARCH_PATH,
+    default_backends: list[str] | None = None,
 ) -> list[BaseSecretsBackend]:
     """
     Initialize secrets backend.
 
     * import secrets backend classes
     * instantiate them and return them in a list
     """
-    backend_list = []
     worker_mode = False
-    if default_backends != DEFAULT_SECRETS_SEARCH_PATH:
+    environment_variable_args: str | None = (
+        "airflow.secrets.environment_variables.EnvironmentVariablesBackend"
+    )
+    metastore_args: str | None = "airflow.secrets.metastore.MetastoreBackend"
+    if default_backends is not None:
         worker_mode = True
+        environment_variable_args = (
+            environment_variable_args if environment_variable_args in default_backends else None
+        )
+        metastore_args = metastore_args if metastore_args in default_backends else None
+    backends_map: dict[str, dict[str, Any]] = {
+        "environment_variable": {
+            "callback": get_importable_secret_backend,
+            "args": (environment_variable_args,),
+        },
+        "metastore": {
+            "callback": get_importable_secret_backend,
+            "args": (metastore_args,),
+        },
+        "custom": {
+            "callback": get_custom_secret_backend,
+            "args": (worker_mode,),
+        },
+    }
 
-    custom_secret_backend = get_custom_secret_backend(worker_mode)
+    backends_order = conf.getlist("secrets", "backends_order", delimiter=",")
 
-    if custom_secret_backend is not None:
-        backend_list.append(custom_secret_backend)
+    required_backends = ["environment_variable"] if worker_mode else ["metastore", "environment_variable"]
+    if missing_backends := [b for b in required_backends if b not in backends_order]:
```

Comment:
Do you think defining an Enum here would make things cleaner?
```
class Backend(Enum):
    ENVIRONMENT_VARIABLE = "environment_variable"
    METASTORE = "metastore"
    CUSTOM = "custom"
```

The hardcoded strings are error prone imo